src/Matcher/ArrayMatcher.php

```diff
@@ -167,6 +167,29 @@ function ($item) use ($skipPattern) {
     }
 
     /**
+     * Finds not existing keys
+     * Excludes keys with pattern which includes Optional Expander
+     *
+     * @param $pattern
+     * @param $values
+     * @return array
+     */
+    private function findNotExistingKeys($pattern, $values)
+    {
+        $notExistingKeys = array_diff_key($pattern, $values);
+
+        return array_filter($notExistingKeys, function ($pattern) use ($values) {
+            try {
+                $typePattern = $this->parser->parse($pattern);
+            } catch (\Exception $e) {
```

Comment:
You're right. What do you think about `Coduo\PHPMatcher\Exception\Exception`? When parsing pattern there could be thrown a `UnknownExpanderException` or `UnknownExpanderException` so we should catch them all.